### DISCUSSION TITLE
Derived atom write function is not firing

### DISCUSSION THREAD

**Discussion by tdronkers04** [pmndrs/jotai#2400] [Q&A] [answered]
Hi-

I'm trying to create an atom `maxCountAtom` that is derived from `countAtom` and `prevMaxAtom`.

`maxCountAtom` should only increment, however with my implementation below, it tracks with `countAtom`, up or down, as long as `countAtom` is positive. I believe my issue is that `prevMaxAtom` is never being updated from the write function of the derived atom.

What am I doing wrong here?
```
const countAtom = atom(0);
const prevMaxAtom = atom(0); // this never gets updated
const maxCountAtom = atom(
  (get) => Math.max(get(countAtom), get(prevMaxAtom)),
  (_get, set, newMax) => set(prevMaxAtom, newMax)
);
```

**Comment by dai-shi** [comment]
Would you like to create a very small reproduction with https://livecodes.io/?x=id/9jbp3dbjsr9 or https://stackblitz.com/edit/jotai-starter please?

**Reply by tdronkers04** [reply]
of course, thank you for taking a look

https://stackblitz.com/edit/jotai-starter-efueog?file=src%2FApp.tsx

**Reply by dai-shi** [reply, marked as answer]
So, it's because you don't call the "write" function:
```js
const [maxCount, setMaxCount] = useAtom(maxCountAtom);
```
Unless you use `setMaxCount`, `prevMaxAtom` value never changes.

I think what you would like to achieve is something different, but at least it's important to understand how the "write" function works, before moving to "Jotai puzzle".
